test/e2e/validation_test.go

```diff
@@ -923,3 +938,27 @@ func finalLine(s string) string {
 	lineSplit := strings.Split(strings.TrimSpace(s), "\n")
 	return strings.TrimSpace(lineSplit[len(lineSplit)-1])
 }
+
+// majorVersion returns the Major portion of the node version annotation
+func majorVersion(node *core.Node) (int, error) {
+	major := semver.Major("v" + node.GetAnnotations()[metadata.VersionAnnotation])
+	if major == "" {
+		return 0, fmt.Errorf("unable to parse major version into semver")
+	}
+	return strconv.Atoi(strings.TrimPrefix(major, "v"))
+}
+
+// nodeShouldHaveCSILabel returns true if a Node on the given platform and WMCO version should be annotated with
+// the CSIConfiguredLabel
+func nodeShouldHaveCSILabel(platform config.PlatformType, wmcoMajorVersion int) bool {
```

Comment:
Nodes on platform? The function name seems to imply that you are looking at a node to decide this.